src/py_zfs_vdev.c

```diff
@@ -0,0 +1,325 @@
+#include "pylibzfs2.h"
+
+#define ZFS_VDEV_STR "<" PYLIBZFS_MODULE_NAME ".ZFSVdev(type=%U, path=%U)>"
+
+static
+PyObject *py_repr_zfs_vdev(PyObject *self) {
+	py_zfs_vdev_t *v = (py_zfs_vdev_t *) self;
+	PyObject *out = PyUnicode_FromFormat(
+		ZFS_VDEV_STR,
+		v->type,
+		v->path ? v->path : py_empty_str()
+	);
+	return (out);
+}
+
+static
+PyObject *py_zfs_vdev_new(PyTypeObject *type, PyObject *args,
+    PyObject *kwds) {
+	py_zfs_vdev_t *self = NULL;
+	self = (py_zfs_vdev_t *)type->tp_alloc(type, 0);
+	return ((PyObject *)self);
+}
+
+static
+int py_zfs_vdev_init(PyObject *type, PyObject *args, PyObject *kwds) {
+	return (0);
+}
+
+static
+void py_zfs_vdev_dealloc(py_zfs_vdev_t *self) {
+	Py_CLEAR(self->parent);
+	Py_CLEAR(self->path);
+	Py_CLEAR(self->type);
+	Py_TYPE(self)->tp_free((PyObject *)self);
+}
+
+static
+PyObject *py_zfs_vdev_asdict(PyObject *self, PyObject *args) {
+	Py_RETURN_NONE;
+}
+
+static
+PyObject *py_zfs_vdev_get_type(py_zfs_vdev_t *self, void *extra) {
+	if (self == NULL || self->type == NULL)
+		Py_RETURN_NONE;
+	return (Py_NewRef(self->type));
+}
+
+static
+PyObject *py_zfs_vdev_get_path(py_zfs_vdev_t *self, void *extra) {
+	if (self == NULL || self->path == NULL)
+		Py_RETURN_NONE;
+	return (Py_NewRef(self->path));
+}
+
+static
+PyObject *py_zfs_vdev_get_name(PyObject *self, PyObject *args) {
+	py_zfs_vdev_t *v = (py_zfs_vdev_t *)self;
+	PyObject *out = NULL;
+	char *name = NULL;
+
+	Py_BEGIN_ALLOW_THREADS
+	PY_ZFS_LOCK(v->pool->pylibzfsp);
+	name = zpool_vdev_name(v->pool->pylibzfsp->lzh, v->pool->zhp,
+	    v->vdev_tree, VDEV_NAME_TYPE_ID);
+	PY_ZFS_UNLOCK(v->pool->pylibzfsp);
+	Py_END_ALLOW_THREADS
+
+	if (name == NULL)
+		PyErr_NoMemory();
```

Comment:
Needs to be `return PyErr_NoMemory();`

> [PyObject](https://docs.python.org/3/c-api/structures.html#c.PyObject) *PyErr_NoMemory()
    Return value: Always NULL. Part of the [Stable ABI](https://docs.python.org/3/c-api/stable.html#stable).

    This is a shorthand for PyErr_SetNone(PyExc_MemoryError); it returns NULL so an object allocation function can write return PyErr_NoMemory(); when it runs out of memory.

Reply:
Thanks, updated.